torchgeo/datasets/utils.py

```diff
@@ -631,6 +639,74 @@ def path_is_vsi(path: Path) -> bool:
     return '://' in str(path) or str(path).startswith('/vsi')
 
 
+def listdir_vfs_recursive(root: Path) -> list[str]:
```

Comment:
Can we make all of these hidden functions (prefixed with an underscore)? They aren't currently documented or exported via `__all__`, but I don't want to take any chances on letting people use these.